server/src/main/java/io/crate/execution/dml/ArrayIndexer.java

```diff
@@ -63,6 +75,16 @@ public void indexValue(List<T> values,
                     );
                 }
             }
+            if (tableVersionCreated.onOrAfter(Version.V_5_9_0)) {
+                // map '[]' to '_array_length_ = 0'
+                // map '[null]' to '_array_length_ = 1'
+                addField.accept(new IntField(arrayLengthFieldName, values.size(), Field.Store.NO));
+            }
+        } else {
+            if (tableVersionCreated.onOrAfter(Version.V_5_9_0)) {
+                // map 'null' to '_array_length_ = -1'
+                addField.accept(new IntField(arrayLengthFieldName, -1, Field.Store.NO));
```

Comment:
In postgres `array_length(NULL) = NULL`

Reply:
ok, [] to 0, NULL to -1 makes sense as long as we construct correct Lucene queries.